### PR TITLE
[BN-1605] scala-3-migration, Do not merge

### PR DESCRIPTION
**Pull request by nandotorterolo**
## Purpose
Scala 3 Strata Node

## Blocks
- (1) sdk cross version accepted and published Sdk - https://github.com/StrataLab/strata-sdk-scala/pull/12
- 
## TODO

- [x]  remove `lib/*` Manual Dependency Management, for sdk, replace when block (1) is solved
- [x]   solve warnings if possible 
- [x]  evantually replace  scalaMock, with   `"org.scalamock" %% "scalamock" % "6.0.0" -> "eu.monniot" %% "scala3mock" % "0.6.5"`  Don't need it, 
- [ ] Discuss about node config, case class +22 fields
- [ ] Pair with Rick, related to missing CI/CD, gh actions, argo, scala 2 -> 3 migration

## Tickets
https://topl.atlassian.net/browse/BN-1605

### CODE REVIEW

**Review by mundacho — APPROVED** [review]
LGTM!

File: models/src/main/scala/org/plasmalabs/models/protocol/ConfigConverter.scala [hunk 1]
```diff
@@ -27,7 +26,7 @@ object ConfigConverter {
     .addModule(module)
     .enable(DeserializationFeature.FAIL_ON_NULL_CREATOR_PROPERTIES)
     .enable(DeserializationFeature.ACCEPT_SINGLE_VALUE_AS_ARRAY)
-    .build() :: ClassTagExtensions) :: ScalaReflectExtensions
+    .build() :: ClassTagExtensions)
```
Comment (o-aleksandrov-topl) [inline models/src/main/scala/org/plasmalabs/models/protocol/ConfigConverter.scala:29]: Do we no longer need ScalaReflectExtension on Scala 3 here?

File: networking/src/main/scala/org/plasmalabs/networking/fsnetwork/BlockDownloadError.scala [hunk 2]
```diff
@@ -19,7 +19,7 @@ object BlockDownloadError {
   object BlockHeaderDownloadError {
 
     case object HeaderNotFoundInPeer extends BlockHeaderDownloadError {
-      override def toString: String = "Block body had been not found in peer"
+      override def toString: String = "Block body had been not found in peer" // TODO should be block header
```
Comment (o-aleksandrov-topl) [inline networking/src/main/scala/org/plasmalabs/networking/fsnetwork/BlockDownloadError.scala:22]: yes shall be "Header" here :)

File: networking/src/main/scala/org/plasmalabs/networking/p2p/FS2P2PServer.scala [hunk 3]
```diff
@@ -29,9 +29,9 @@ object FS2P2PServer {
     ed25519Resource:         Resource[F, Ed25519]
   ): Resource[F, P2PServer[F]] =
     for {
-      implicit0(logger: Logger[F]) <- Slf4jLogger.fromName("Node.P2P").toResource
-      _                            <- eventLogger(peersStatusChangesTopic)
-      sockets                      <- socketsStream[F](host, port)
+      given Logger[F] <- Slf4jLogger.fromName("Bifrost.P2P").toResource
```
Comment (o-aleksandrov-topl) [inline networking/src/main/scala/org/plasmalabs/networking/p2p/FS2P2PServer.scala:32]: shall be "Node" not Bifrost

**Review by o-aleksandrov-topl — APPROVED** [review]

**Pull request merged**